google-beta/resource_pubsub_subscription.go

```diff
@@ -16,213 +34,429 @@ func resourcePubsubSubscription() *schema.Resource {
 				Optional: true,
-				Computed: true,
-				ForceNew: true,
 			},
-
-			"path": {
-				Type:     schema.TypeString,
-				Computed: true,
+			"labels": {
+				Type:     schema.TypeMap,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
 			},
-
 			"push_config": {
 				Type:     schema.TypeList,
 				Optional: true,
 				MaxItems: 1,
 				Elem: &schema.Resource{
 					Schema: map[string]*schema.Schema{
+						"push_endpoint": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
 						"attributes": {
 							Type:     schema.TypeMap,
 							Optional: true,
 							Elem:     &schema.Schema{Type: schema.TypeString},
 						},
-
-						"push_endpoint": {
-							Type:     schema.TypeString,
-							Required: true,
-						},
 					},
 				},
 			},
+			"path": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"project": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
 		},
 	}
 }
 
 func resourcePubsubSubscriptionCreate(d *schema.ResourceData, meta interface{}) error {
 	config := meta.(*Config)
 
-	project, err := getProject(d, config)
+	obj := make(map[string]interface{})
+	nameProp, err := expandPubsubSubscriptionName(d.Get("name"), d, config)
 	if err != nil {
 		return err
+	} else if v, ok := d.GetOkExists("name"); !isEmptyValue(reflect.ValueOf(nameProp)) && (ok || !reflect.DeepEqual(v, nameProp)) {
+		obj["name"] = nameProp
+	}
+	topicProp, err := expandPubsubSubscriptionTopic(d.Get("topic"), d, config)
+	if err != nil {
+		return err
+	} else if v, ok := d.GetOkExists("topic"); !isEmptyValue(reflect.ValueOf(topicProp)) && (ok || !reflect.DeepEqual(v, topicProp)) {
+		obj["topic"] = topicProp
+	}
+	labelsProp, err := expandPubsubSubscriptionLabels(d.Get("labels"), d, config)
+	if err != nil {
+		return err
+	} else if v, ok := d.GetOkExists("labels"); !isEmptyValue(reflect.ValueOf(labelsProp)) && (ok || !reflect.DeepEqual(v, labelsProp)) {
+		obj["labels"] = labelsProp
+	}
+	pushConfigProp, err := expandPubsubSubscriptionPushConfig(d.Get("push_config"), d, config)
+	if err != nil {
+		return err
+	} else if v, ok := d.GetOkExists("push_config"); !isEmptyValue(reflect.ValueOf(pushConfigProp)) && (ok || !reflect.DeepEqual(v, pushConfigProp)) {
+		obj["pushConfig"] = pushConfigProp
+	}
+	ackDeadlineSecondsProp, err := expandPubsubSubscriptionAckDeadlineSeconds(d.Get("ack_deadline_seconds"), d, config)
+	if err != nil {
+		return err
+	} else if v, ok := d.GetOkExists("ack_deadline_seconds"); !isEmptyValue(reflect.ValueOf(ackDeadlineSecondsProp)) && (ok || !reflect.DeepEqual(v, ackDeadlineSecondsProp)) {
+		obj["ackDeadlineSeconds"] = ackDeadlineSecondsProp
 	}
 
-	name := getComputedSubscriptionName(project, d.Get("name").(string))
-	computed_topic_name := getComputedTopicName(project, d.Get("topic").(string))
-
-	//  process optional parameters
-	var ackDeadlineSeconds int64
-	ackDeadlineSeconds = 10
```

Comment:
Does this default get preserved? I'd _imagine_ it's the API default so we're fine, but if not we will sadly probably need to include it still.